samza-azure/src/main/java/org/apache/samza/system/azureblob/producer/AzureBlobSystemProducer.java

```diff
@@ -0,0 +1,505 @@
+    ReadWriteLock lock = sourceSendFlushLockMap.get(source);
+    if (lock == null) {
+      throw new SystemProducerException("Attempting to flush source: " + source + " but it was not registered");
+    }
+    lock.writeLock().lock();
+    Map<String, AzureBlobWriter> sourceWriterMap = writerMap.get(source);
+    try {
+      // first flush all the writers
+      // then close and remove all the writers
+      flushWriters(sourceWriterMap);
+      closeWriters(source, sourceWriterMap);
+    } catch (Exception e) {
+      metrics.updateErrorMetrics(source);
+      throw new SystemProducerException("Flush failed for system:" + systemName + " and source: " + source, e);
+    } finally {
+      sourceWriterMap.clear();
+      lock.writeLock().unlock();
+    }
+  }
+
+  @VisibleForTesting
+  void setupAzureContainer(String accountName, String accountKey) {
+    try {
+      // Use your Azure Blob Storage account's name and key to create a credential object to access your account.
+      StorageSharedKeyCredential credential = new StorageSharedKeyCredential(accountName, accountKey);
+
+      HttpClient httpClient;
+      if (config.getUseProxy(systemName)) {
+        LOG.info("HTTP Proxy setup for AzureBlob pipeline");
+        httpClient = new NettyAsyncHttpClientBuilder()
+            .proxy(new ProxyOptions(ProxyOptions.Type.HTTP,
+            new InetSocketAddress(config.getAzureProxyHostname(systemName), config.getAzureProxyPort(systemName)))).build();
+      } else {
+        httpClient = HttpClient.createDefault();
+      }
+
+      // From the Azure portal, get your Storage account blob service AsyncClient endpoint.
+      String endpoint = String.format(Locale.ROOT, AZURE_URL, accountName);
+
+      HttpLogOptions httpLogOptions = new HttpLogOptions();
+      httpLogOptions.setLogLevel(HttpLogDetailLevel.BASIC);
+      BlobServiceAsyncClient storageClient =
+          new BlobServiceClientBuilder()
+          .httpLogOptions(httpLogOptions)
+          .endpoint(endpoint)
+          .credential(credential)
+          .httpClient(httpClient)
+          .buildAsyncClient();
+
+
+      SkuName accountType = storageClient.getAccountInfo().block().getSkuName();
+      long flushThresholdSize = config.getMaxFlushThresholdSize(systemName);
+      boolean isPremiumAccount = SkuName.PREMIUM_LRS == accountType;
+      if (isPremiumAccount && flushThresholdSize > PREMIUM_MAX_BLOCK_SIZE) { // 100 MB
+        throw new SystemProducerException("Azure storage account with name: " + accountName
+            + " is a premium account and can only handle upto 100MB threshold size. Given flush threshold size is "
+            + flushThresholdSize);
+      } else if (!isPremiumAccount && flushThresholdSize > STANDARD_MAX_BLOCK_SIZE) { // STANDARD account
+        throw new SystemProducerException("Azure storage account with name: " + accountName
+            + " is a standard account and can only handle upto 4MB threshold size. Given flush threshold size is "
```

Comment:
Should the `100MB` and `4MB` be changed to use `PREMIUM_MAX_BLOCK_SIZE` and `STANDARD_MAX_BLOCK_SIZE`? Then, if someone changes the max block size constants, the log messages will be updated automatically.